--- cpp/ggml-alloc.c	2024-10-24 14:42:04.320406093 +0800
+++ cpp/ggml-alloc.c	2024-10-24 15:49:02.307240277 +0800
@@ -272,22 +272,20 @@
 
     struct ggml_allocr * alloc = (struct ggml_allocr *)malloc(sizeof(struct ggml_allocr));
 
-    *alloc = (struct ggml_allocr){
-        /*.buffer        = */ buffer,
-        /*.buffer_owned  = */ true,
-        /*.base          = */ ggml_backend_buffer_get_base(buffer),
-        /*.alignment     = */ alignment,
-        /*.n_free_blocks = */ 0,
-        /*.free_blocks   = */ {{0}},
-        /*.hash_table    = */ {{0}},
-        /*.max_size      = */ 0,
-        /*.measure       = */ false,
-        /*.parse_seq     = */ {0},
-        /*.parse_seq_len = */ 0,
+    alloc->buffer = buffer;
+    alloc->buffer_owned = true;
+    alloc->data = ggml_backend_buffer_get_base(buffer);
+    alloc->alignment = alignment;
+    alloc->n_free_blocks = 0;
+    memset(alloc->free_blocks, 0, sizeof(alloc->free_blocks));
+    memset(alloc->hash_table, 0, sizeof(alloc->hash_table));
+    alloc->max_size = 0;
+    alloc->measure = false;
+    memset(alloc->parse_seq, 0, sizeof(alloc->parse_seq));
+    alloc->parse_seq_len = 0;
 #ifdef GGML_ALLOCATOR_DEBUG
-        /*.allocated_tensors = */ {0},
+    memset(alloc->allocated_tensors, 0, sizeof(alloc->allocated_tensors));
 #endif
-    };
 
     ggml_allocr_reset(alloc);
 
@@ -304,22 +302,20 @@
 struct ggml_allocr * ggml_allocr_new_from_buffer(struct ggml_backend_buffer * buffer) {
     struct ggml_allocr * alloc = (struct ggml_allocr *)malloc(sizeof(struct ggml_allocr));
 
-    *alloc = (struct ggml_allocr){
-        /*.buffer        = */ buffer,
-        /*.buffer_owned  = */ false,
-        /*.base          = */ ggml_backend_buffer_get_base(buffer),
-        /*.alignment     = */ ggml_backend_buffer_get_alignment(buffer),
-        /*.n_free_blocks = */ 0,
-        /*.free_blocks   = */ {{0}},
-        /*.hash_table    = */ {{0}},
-        /*.max_size      = */ 0,
-        /*.measure       = */ false,
-        /*.parse_seq     = */ {0},
-        /*.parse_seq_len = */ 0,
+    alloc->buffer = buffer;
+    alloc->buffer_owned = false;
+    alloc->data = ggml_backend_buffer_get_base(buffer);
+    alloc->alignment = ggml_backend_buffer_get_alignment(buffer);
+    alloc->n_free_blocks = 0;
+    memset(alloc->free_blocks, 0, sizeof(alloc->free_blocks));
+    memset(alloc->hash_table, 0, sizeof(alloc->hash_table));
+    alloc->max_size = 0;
+    alloc->measure = false;
+    memset(alloc->parse_seq, 0, sizeof(alloc->parse_seq));
+    alloc->parse_seq_len = 0;
 #ifdef GGML_ALLOCATOR_DEBUG
-        /*.allocated_tensors = */ {0},
+    memset(alloc->allocated_tensors, 0, sizeof(alloc->allocated_tensors));
 #endif
-    };
 
     ggml_allocr_reset(alloc);
 
